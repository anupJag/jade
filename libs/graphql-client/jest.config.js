module.exports = {
  name: 'ui-graphql-client',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': ['babel-jest', { cwd: __dirname, configFile: './babel-jest.config.json' }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/ui/graphql-client',
};
