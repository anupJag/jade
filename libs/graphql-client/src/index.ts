export * from './lib/graphqlClient';
export * from './lib/contentfulGraphqlClient';
export * from './lib/queries';
export * from './lib/authorisedGraphqlClient';
