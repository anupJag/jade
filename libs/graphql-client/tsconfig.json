{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react",
    "allowJs": false,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "resolveJsonModule": true,
    "types": ["node", "jest"]
  },
  "files": [
    "../../../node_modules/@nrwl/react/typings/cssmodule.d.ts",
    "../../../node_modules/@nrwl/react/typings/image.d.ts"
  ],
  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"]
}
