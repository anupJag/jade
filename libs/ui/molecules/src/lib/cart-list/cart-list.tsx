import React, { FC, useState } from 'react';
import { Box, Image } from '@chakra-ui/core';
import Grid from '@material-ui/core/Grid';
import AddIcon from '@material-ui/icons/Add';
import RemoveIcon from '@material-ui/icons/Remove';
import DeleteIcon from '@material-ui/icons/Delete';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import makeStyles from '@material-ui/core/styles/makeStyles';
import { green } from '@material-ui/core/colors';
import { CartListContainer } from './cart-list.styled';
import { AddToCart } from '../add-to-cart/add-to-cart';

export const useStyles = makeStyles(theme => ({
  qunaity: {
    border: '1px solid #cccccc',
  },
}));

type Props = {
  quantity: any;
  product?: any;
  addToCart?: any;
};

const Quanity: FC<Props> = ({ product, quantity, addToCart }) => {
  const classes = useStyles();
  const [qty, setQty] = useState(quantity);

  const addQuantity = () => {
    setQty(q => {
      addToCart({ quantity: q + 1, productId: product.id });
      return q + 1;
    });
  };

  const removeQuantity = () => {
    setQty(q => {
      const count = q - 1;
      if (count < 1) {
        return 1;
      }
      addToCart({ quantity: count, productId: product.id });
      return count;
    });
  };
  return (
    <Grid container alignItems="baseline">
      <Grid item xs={4}>
        <Typography component="span" variant="subtitle2" color="textPrimary">
          Quantity
        </Typography>
      </Grid>
      <Grid item xs={2}>
        <IconButton onClick={addQuantity}>
          <AddIcon fontSize="small" style={{ color: green[500] }} />
        </IconButton>
      </Grid>
      <Grid item xs={2} container justify="center" className={classes.qunaity}>
        <Typography component="span" variant="body2" color="textPrimary">
          {qty}
        </Typography>
      </Grid>
      <Grid item xs={2}>
        <IconButton onClick={removeQuantity}>
          <RemoveIcon fontSize="small" color="secondary" />
        </IconButton>
      </Grid>
    </Grid>
  );
};

// TODO :
// eslint-disable-next-line @typescript-eslint/no-empty-function
export const CartList = ({ data = [], onDelete = () => {} }: any) => {
  return data.map(({ id, quantity, product = {} }: any) => {
    const srcProductImages = product.images.filter(img => img.isDefault === true);
    const defaultImage = srcProductImages && srcProductImages[0];

    return (
      <CartListContainer mt="2" d="flex" overflow="hidden" key={id}>
        <Image size="100px" objectFit="cover" src={defaultImage.url} alt={defaultImage.altText} />
        <Box p={2} ml={2} flex="1" width="1%">
          <Box mt="1" color="blue.500" as="h6" lineHeight="tight" isTruncated>
            {product.title}
          </Box>
          <Box
            mt="1"
            color="gray.500"
            fontWeight="normal"
            letterSpacing="wide"
            fontSize="xs"
            textTransform="capitalize">
            <strong>Price:</strong> {product.pricing && product.pricing.list} &nbsp;
          </Box>
          <AddToCart>
            <Quanity product={product} quantity={quantity} />
          </AddToCart>
        </Box>
        <IconButton
          disableFocusRipple
          disableRipple
          edge="end"
          aria-label="delete"
          onClick={() => onDelete({ id })}>
          <DeleteIcon color="secondary" fontSize="small" />
        </IconButton>
      </CartListContainer>
    );
  });
};
