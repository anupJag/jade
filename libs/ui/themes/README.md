# fe-themes

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test fe-themes` to execute the unit tests via [Jest](https://jestjs.io).
