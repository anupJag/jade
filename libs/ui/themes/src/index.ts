export * from './lib/defaultTheme';
export * from './lib/styled';
export * from './lib/themeProvider';
export { default as styled } from './lib/styled';
export * from './lib/GlobalStyles';
