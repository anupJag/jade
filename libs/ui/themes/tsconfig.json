{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react",
    "allowJs": false,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "types": ["node", "jest"]
  },
  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"]
}
