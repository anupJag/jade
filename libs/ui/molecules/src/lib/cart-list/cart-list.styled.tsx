import { styled } from '@jade/ui/themes';
import { Box } from '@chakra-ui/core';

export const CartListContainer = styled(Box)`
  border-bottom: 1px solid #dcdcdc;
  padding-bottom: 5px;
  padding-top: 5px;
`;
