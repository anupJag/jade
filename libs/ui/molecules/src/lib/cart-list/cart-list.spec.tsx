import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@jade/ui/themes';
import { ApolloProvider } from '@apollo/react-hooks';
import { initializeApollo } from '@jade/graphql-client';
import { CartList } from './cart-list';

describe('<CartList />', () => {
  let CartListComponent;
  const initApolloClient = initializeApollo();

  const onDelete = jest.fn();
  const data = [
    {
      id: 'test_1234',
      quantity: 10,
      product: {
        id: 'product_123',
        defaultImageUrl: '/product.jpg',
        defaultImageAlt: 'Product Alt',
        price: '120',
        title: 'Product Title',
      },
    },
  ];
  let container;
  let ListEl;

  beforeEach(() => {
    CartListComponent = render(
      <ApolloProvider client={initApolloClient}>
        <ThemeProvider>
          <CartList onDelete={onDelete} data={data} />
        </ThemeProvider>
      </ApolloProvider>,
    );

    ({ container } = CartListComponent);
  });

  /**
   * Capturing Sanpshot : CartList
   *
   *  To check style / structure changes in Styled Components
   */
  test('snapshot check', () => {
    const { asFragment } = CartListComponent;
    expect(asFragment()).toMatchSnapshot();
  });
});
