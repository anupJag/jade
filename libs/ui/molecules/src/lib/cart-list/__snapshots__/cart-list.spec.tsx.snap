// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartList /> snapshot check 1`] = `<DocumentFragment />`;
